coeus-impl/src/main/java/org/kuali/coeus/common/budget/impl/period/BudgetPeriodRule.java

```diff
@@ -340,6 +340,10 @@ private boolean isValidNewBudgetPeriod(Budget budget, BudgetPeriod newBudgetPeri
                         newBudgetPeriod.setBudgetPeriod(periodNum+1);
                     }
                     break;
+                }else if(newPeriodStartDate.compareTo(periodEndDate) <= 0) {
+                    dateCompareValue = "ERROR_NEW_PERIOD_START_BEFORE_PREVIOUS_END";
+                    saveErrors(dateCompareValue, errorMap, newPeriodDateParams);
+                    validNewBudgetPeriod = false;
```

Comment:
This doesn't seem correct to me. This is in a loop of all existing periods. If I've modified dates and I'm trying to insert a period between 2 existing periods, seems as though this will always throw an error incorrectly in that case.